### ISSUE TITLE
Implement modular inverse

### ISSUE THREAD

**Issue by dignifiedquire** [rust-num/num-bigint#60]
I had the need for this and have a variant that works well for positive numbers here, if there is interest: https://github.com/dignifiedquire/rust-rsa/blob/master/src/math.rs#L118

**Comment by dmhacker** [comment]
I second this.

**Comment by cuviper** [comment]
This will probably be met generically by rust-num/num-integer#10.

**Comment by mikong** [comment]
@dignifiedquire, your link is currently redirecting then returning 404 for me. The RustCrypto/RSA repo doesn't have a src/math.rs, but I can see that it uses a `ModInverse` trait and that the trait is implemented in your fork of num-bigint. Is that where you'd like us to look at?

**Comment by dignifiedquire** [comment]
yes, that code moved out of the RSA library into my fork of num-bigint into: https://github.com/dignifiedquire/num-bigint/blob/master/src/algorithms/mod_inverse.rs
